File: Java/TravelServices/src/main/java/travel/services/ITravelServices.java
package travel.services;

import travel.model.Employee;
import travel.model.Flight;
import travel.model.Ticket;

import java.time.LocalDate;
import java.util.Collection;

public interface ITravelServices {
    void login(Employee employee, ITravelObserver client) throws TravelException;
    void logout(Employee employee, ITravelObserver client) throws TravelException;
    Collection<Flight> filterFlightsByDestinationAndDeparture(String destination, LocalDate departure) throws TravelException;
    Collection<Flight> filterFlightByAvailableSeats() throws TravelException;
    void addTicket(Ticket ticket) throws TravelException;
    void addFlight(Flight flight) throws TravelException;
}

File: ProiectCS/travelServices/ITravelServices.cs
using System;
using System.Collections.Generic;
using travelModel;

namespace travelServices
{

    public interface ITravelServices
    {
        void login(Employee employee, ITravelObserver client);
        void logout(Employee employee, ITravelObserver client);
        IEnumerable<Flight> filterFlightsByDestinationAndDeparture(String destination, DateTime departure);
        IEnumerable<Flight> filterFlightByAvailableSeats();
        void addTicket(Ticket ticket);
    }
}